src/osd/PGLog.cc

```diff
@@ -805,15 +787,23 @@ void PGLog::_write_log_and_missing(
   bool *rebuilt_missing_with_deletes, // in/out param
   set<string> *log_keys_debug
   ) {
-  set<string> to_remove(trimmed_dups);
-  for (set<eversion_t>::const_iterator i = trimmed.begin();
-       i != trimmed.end();
-       ++i) {
-    to_remove.insert(i->get_key_name());
-    if (log_keys_debug) {
-      assert(log_keys_debug->count(i->get_key_name()));
-      log_keys_debug->erase(i->get_key_name());
+  set<string> to_remove;
+  if (trimmed_dups) {
+    to_remove.swap(*trimmed_dups);
```

Comment:
this changes the semantic of this method. before this change, `trimmed` and `trimmed_dups` are not changed after `_write_log_and_missing()` returns. after this change, both of them are empty. is this expected?

Reply:
yes, that's expected. they aren't used any more

Reply:
if that's the case, i need to take a closer look.

Reply:
and i'd suggest update the commit message accordingly.

Reply:
@tchaikov - updated

Reply:
nit, could just 
```c++
to_remove = std::move(*trimmed_dups);
```

as we don't care about `trimmed_dups` anymore after this call.

Reply:
IMO it's better to leave a container in a consistent(=empty) state to avoid potential issues if code evolves.